src/Deposit.js

```diff
@@ -1014,16 +1004,23 @@ export default class Deposit {
    * so a deposit cannot start auto-submitting and then switch to auto-minting
    * mid-stream---whichever is called first will be the flow that will occur.
    *
+   * @param {OnErrorHandler} [onError] A handler that receives an error (optional).
+   *
    * @return {AutoSubmitState} An object with promises to various stages of
    *         the auto-submit lifetime. Each promise can be fulfilled or
    *         rejected, and they are in a sequence where later promises will be
    *         rejected by earlier ones.
    */
-  autoSubmit() {
+  autoSubmit(onError) {
     // Only enable auto-submitting once.
     if (this.autoSubmittingState) {
       return this.autoSubmittingState
     }
+
+    if (onError) {
+      this.eventEmitter.on("error", onError)
+    }
```

Comment:
This isn't what I'd call the expected behavior, fwiw, because this keeps the error handler registered. Passing it to the function implies a single-shot usage. You could use `.one` here, but if there's no error invoked later on then that could lead to an unexpected call with a future error event.

I would just invoke this in the same place we trigger the event below, if it's defined.